src/Arcus.EventGrid.Testing/Infrastructure/Hosts/EventConsumerHost.cs

```diff
@@ -43,10 +43,10 @@ protected static void EventsReceived(string rawReceivedEvents, ILogger logger)
             JArray parsedEvents = JArray.Parse(rawReceivedEvents);
             foreach (JToken parsedEvent in parsedEvents)
             {
-                string eventId = parsedEvent["Id"]?.ToString();
+                string eventId = DetermineEventId(parsedEvent);
                 if (eventId == null)
                 {
-                    logger.LogWarning("Event was received without an event id. Payload : {receivedEvent}", parsedEvent);
+                    logger.LogWarning($"Event was received without an event id. Payload : {parsedEvent}");
```

Comment:
Maybe pass-allong the argument as a logging argument instead of concat it yourself?

Reply:
The issue is that with that approach it does not provide a valid json in the output, similar to this: https://github.com/arcus-azure/arcus.eventgrid/pull/70#issuecomment-478965852